### PR TITLE
NormalizationInverse class

### PR DESCRIPTION
**Pull request by EnricoTrizio**
## Description
Implements a helper function to return the inverse of a Normalization object as a torch.nn.Module.

Added return_decoder_model method to autoencoder-based CVs (AE_CV and VAE_CV)

## Status
- [ ] Ready to go

### CODE REVIEW

**Review by andrrizzi — APPROVED** [review]
Looks good to me!

I'm not sure about the use case for this, so I'll just throw a couple of ideas.

If the most frequent use case is to access the decoder with the inverse normalization (when present), we could think of moving the attribute ``self.decoder`` -> ``self._decoder``, and implement ``return_decoder_model`` as the property ``def decoder``. If we want to support accessing the decoder without the normalization, we can also implement a getter ``def get_decoder(return_normalization=True/False)``.

Another idea I'm not sure we want to implement, is to implement the inverse as a wrapper.
```python
class Inverse(torch.nn.Module):
    def __init__(module):
        self.module = module
    def forward(*args, **kwargs):
        return self.module.inverse(*args, **kwargs)
    def inverse(*args, **kwargs):
        return super().forward(*args, **kwargs)
```
the advantage here would be that if the normalization layer (or any other invertible layer) is updated during training, the inverse is updated automatically as it shares the same parameters so we could use it also directly inside our CV models.

File: mlcolvar/cvs/unsupervised/autoencoder.py [hunk 1]
```diff
@@ -104,6 +104,15 @@ def training_step(self, train_batch, batch_idx):
         name = 'train' if self.training else 'valid'       
         self.log(f'{name}_loss', loss, on_epoch=True)
         return loss
+    
+    def return_decoder_model(self):
```
[the remark this answers is not in the document]
Comment (EnricoTrizio) [reply mlcolvar/cvs/unsupervised/autoencoder.py:108]: Ok perfect

File: mlcolvar/core/transform/normalization.py [hunk 2]
```diff
@@ -176,6 +176,23 @@ def inverse(self, x: torch.Tensor) -> (torch.Tensor):
...
+                     **args):
+            """Apply the inverse of a Normalization object as an torch.nn.Module.
+
+            Parameters
+            ----------
+            norm : torch.nn.Module
+                Normalization object to be inversed
+            """
+            super().__init__(**args)
+            self.set_custom(mean = norm.mean, range = norm.range)
```
[the remark this answers is not in the document]
Comment (EnricoTrizio) [reply mlcolvar/core/transform/normalization.py:192]: Replaced `NormalizationInverse` class with `Inverse` class in `transform.utils`

**Review by luigibonati — APPROVED** [review]
sounds all good to me! I like the wrapper class since, in general, it might be something applicable also to the other transforms!
